common/network-types/src/types.rs

```diff
@@ -86,6 +86,14 @@ impl IpOrHost {
         self.resolve(resolver).await
     }
 
+    #[cfg(all(test, feature = "runtime-async-std"))]
```

Comment:
Would be good to explain the reasoning why is this needed.

Reply:
done